api/v2/serializers/details/credential.py

```diff
@@ -8,7 +11,34 @@ class CredentialSerializer(serializers.HyperlinkedModelSerializer):
     url = UUIDHyperlinkedIdentityField(
         view_name='api:v2:credential-detail',
     )
-    identity = IdentitySummarySerializer()
+    identity = IdentitySummarySerializer(read_only=True)
+
+    def validate_text(self, key, value):
+        """
+        """
+        return value
+
+    def validate_url(self, key, url_value):
```

Comment:
After looking at the two serializers, there is no "great" way to encapsulate the behavior within a common serializer (due to them being model-specific keys and properties).

Therefore, i'm willing to accept the additional 'weight' here.